x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/service/CachingServiceAccountTokenStoreTests.java

```diff
@@ -53,22 +55,30 @@ public void stop() {
         }
     }
 
+    private ServiceAccountToken newMockServiceAccountToken(ServiceAccountId accountId, String tokenName, SecureString secret) {
```

Comment:
Added this mocking instead of making the `ServiceAccountToken` constructor public. The `ServiceAccountToken` needs to be in core since it's part of the security extension interface. The `ServiceAccountService` could also be moved to core, but that involves moving all service account definitions to core, which feels like too big of a refactor to justify.